cni/pkg/nftables/host_sets.go

```diff
@@ -0,0 +1,385 @@
+			return nil
+		}
+		setName = h.v6SetName
+	}
+
+	nft, err := h.nftProvider(knftables.InetFamily, AmbientNatTable)
+	if err != nil {
+		return err
+	}
+
+	tx := nft.NewTransaction()
+
+	// Add element to the set
+	tx.Add(&knftables.Element{
+		Set:     setName,
+		Key:     []string{ipToInsert.String()},
+		Comment: knftables.PtrTo(comment),
+	})
+
+	if err := nft.Run(context.TODO(), tx); err != nil {
+		return fmt.Errorf("failed to add IP %s to nftables set %s: %w", ip, setName, err)
+	}
+
+	builder.LogNftRules(tx)
+	return nil
+}
+
+// DeleteIP removes an IP address from the appropriate set
+func (h *HostNftSetManager) DeleteIP(ip netip.Addr, ipProto uint8) error {
+	nft, err := h.nftProvider(knftables.InetFamily, AmbientNatTable)
+	if err != nil {
+		return err
+	}
+
+	ipToDel := ip.Unmap()
+	setName := h.v4SetName
+	if ipToDel.Is6() {
+		if !h.enableIPv6 {
+			log.Debugf("IPv6 is not enabled. Skipping the deletion of IP: %v", ip)
+			return nil
+		}
+		setName = h.v6SetName
+	}
+
+	tx := nft.NewTransaction()
+
+	// Delete element from the set
+	tx.Delete(&knftables.Element{
+		Set: setName,
+		Key: []string{ipToDel.String()},
+	})
+
+	if err := nft.Run(context.TODO(), tx); err != nil {
+		return fmt.Errorf("failed to delete IP %s from the nftables set %s: %w", ip, setName, err)
+	}
+
+	builder.LogNftRules(tx)
+	return nil
+}
+
+// Flush clears all entries from both sets
+func (h *HostNftSetManager) Flush() error {
+	nft, err := h.nftProvider(knftables.InetFamily, AmbientNatTable)
+	if err != nil {
+		return err
+	}
+
+	tx := nft.NewTransaction()
+
+	// Flush IPv4 set
+	tx.Flush(&knftables.Set{
+		Name: h.v4SetName,
+	})
+
+	// Flush IPv6 set if enabled
+	if h.enableIPv6 {
+		tx.Flush(&knftables.Set{
+			Name: h.v6SetName,
+		})
+	}
+
+	if err := nft.Run(context.TODO(), tx); err != nil {
+		return fmt.Errorf("failed to flush nftables sets: %w", err)
+	}
+
+	builder.LogNftRules(tx)
+	return nil
+}
+
+// DestroySet completely destroys both sets and the table
+func (h *HostNftSetManager) DestroySet() error {
+	nft, err := h.nftProvider(knftables.InetFamily, AmbientNatTable)
+	if err != nil {
+		return err
+	}
+
+	tx := nft.NewTransaction()
+
+	// Delete the entire table (which will delete sets and rules)
+	tx.Delete(&knftables.Table{})
+
+	if err := nft.Run(context.TODO(), tx); err != nil {
+		return fmt.Errorf("failed to destroy nftables table %s: %w", AmbientNatTable, err)
+	}
+
+	builder.LogNftRules(tx)
+	return nil
+}
+
+// ClearEntriesWithComment removes all entries with the specified comment
+func (h *HostNftSetManager) ClearEntriesWithComment(comment string) error {
+	nft, err := h.nftProvider(knftables.InetFamily, AmbientNatTable)
+	if err != nil {
+		return err
+	}
+
+	// A pod can have multiple IPs, so we have to clear it from both v4 and v6 sets.
+	if err := h.clearEntriesFromSetWithComment(nft, h.v4SetName, comment); err != nil {
+		return fmt.Errorf("failed to clear entries from IPv4 address set: %w", err)
```

Comment:
If there's an issue with clearing the v4 table, we never clear the v6 table which could create stale entries in the v6 table. We probably want to collect the errors from both and then return